src/main/java/com/entropyteam/entropay/common/BaseService.java

```diff
@@ -111,12 +113,19 @@ private Collection<Predicate> buildIdPredicates(Root<Entity> root, Filter filter
     }
 
     private Collection<Predicate> buildEntityPredicates(Root<Entity> root, Filter filter, CriteriaBuilder cb) {
-        if (MapUtils.isEmpty(filter.getGetByFieldsFilter())) {
+        if (MapUtils.isEmpty(filter.getGetByFieldsFilter()) && MapUtils.isEmpty(filter.getGetByDateFieldsFilter())) {
             return CollectionUtils.emptyCollection();
         }
         Collection<Predicate> predicates = filter.getGetByFieldsFilter().entrySet().stream().filter(f -> f.getKey() != SEARCH_TERM_KEY)
                 .map(f -> cb.equal(root.get(f.getKey()), f.getValue())).collect(Collectors.toSet());
 
+        if(filter.getGetByDateFieldsFilter().containsKey(DATE_TERM_KEY)){
+            String year = filter.getGetByDateFieldsFilter().get(DATE_TERM_KEY);
+            LocalDate dateFrom = LocalDate.parse(year+"-01-01");
+            LocalDate dateTo = LocalDate.parse(year+"-12-31");
+            Predicate predicate = cb.between(root.get("date"), dateFrom,dateTo);
```

Comment:
This filter will work only in tables that has `date` column. you should implement something like `getColumnsForSearch` as we did for the search term filter